tests/repl/test_engine.rs

```diff
@@ -52,6 +52,21 @@ fn in_and_if_else() -> TestResult {
     )
 }
 
+#[test]
+fn in_with_closure() -> TestResult {
+    // Can use $in twice
+    run_test(r#"3 | do { let x = $in; let y = $in; $x + $y }"#, "6")
```

Comment:
Maybe worth adding another test à la

```nushell
> 3 | do { let x = $in; let y = $in; $x + $y | $in * 4 } 
24
```

to ensure that in the next pipeline element `$in` refers to the previous pipeline step